### PR TITLE
TST: `test_highlight.py` convert to functional tests not class

### PR DESCRIPTION
**Pull request by attack68**
- [x] add `pytest.fixture` instead of Class setup method
- [x] no test removed or added, just amended to accept new fixtures

### CODE REVIEW

File: pandas/tests/io/formats/style/test_highlight.py [hunk 1]
```diff
@@ -5,61 +5,68 @@
...
+        (1, 0): [("background-color", "red")],
+        (1, 1): [("background-color", "red")],
+    }
+    assert result == expected
+
+
+def test_highlight_null_subset(s):
+    # GH 31345
+    result = (
+        s.highlight_null(null_color="red", subset=["A"])
```
Comment (jreback) [inline pandas/tests/io/formats/style/test_highlight.py:33]: nit: can you line these up on the periods

File: pandas/tests/io/formats/style/test_highlight.py [hunk 2]
```diff
@@ -5,61 +5,68 @@
...
-            (1, 0): [("background-color", "red")],
-        }
-        if f == "highlight_min":
-            df = -self.df
-        else:
-            df = self.df
-        result = getattr(df.style, f)(axis=1, color="red")._compute().ctx
-        assert result == expected
+@pytest.fixture
+def s(df):
```
Comment (jreback) [inline pandas/tests/io/formats/style/test_highlight.py:17]: can you call this maybe styler, s is too short

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/formats/style/test_highlight.py (hunk 1 above)
Comment (attack68) [reply pandas/tests/io/formats/style/test_highlight.py]: `black` does the aligning here, not me.

File: pandas/tests/io/formats/style/test_highlight.py (hunk 2 above)
Comment (attack68) [reply pandas/tests/io/formats/style/test_highlight.py]: done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @attack68

**Pull request merged**